src/inc/msquic_posix.h

```diff
@@ -186,7 +186,11 @@ typedef union QUIC_ADDR {
 extern "C" {
 #endif
 
+#if defined(CX_PLATFORM_DARWIN)
+#define QUIC_LOCALHOST_FOR_AF(Af) ((Af == QUIC_ADDRESS_FAMILY_INET) ? "localhost" : "::1")
```

Comment:
Its `ip6-localhost` not valid on macOS? Interesting, but not super shocking.

Reply:
no. it seems like on linux it is in /etc/hosts but not macOS. This can be also distro specific.

Reply:
Since this is a public header, I wonder if we should change these on all posix platforms to use `127.0.0.1` for IPv4 and `::1` for IPv6. Since those actually are standardized. @nibanks opinion?

Reply:
Or at minimum, switch ipv6 to `::1` on both platforms.

Reply:
Is it public? I could not find any reference to this besides tests. 'localhost' is pretty standard IMHO but I don't think it its guarantied to exist. In case of the test, you would get name resolution failure - and that can generally happen for any name IMHO. 
'localhost' can also resolve to ::1 on certain systems:
https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html

Reply:
The header resolves publicly, but its probably not too big of a deal. At least for IPv4 leaving localhost is fine, but I think ::1 would be better for IPv6 since that is more likely to work across distros. I don't want to get into a case where our tests don't work on certain distros, and I'd like to avoid the ifdef if possible too.

Reply:
Don't change it to an IP address. The whole point it to check name resolution as well.

Reply:
So then what do we do? If ip6-localhost isn't in hosts on macOS, do we need to modify the tests to configure that? That's difficult to do without requiring admin.

Reply:
So neither `localhost` or `ip6-localhost` work on mac for v6?

Reply:
It looks like localhost does, but not ip6-localhost

Reply:
If `localhost` works for both v4 & v6, then just use that. That's how it works on Windows.

Reply:
I did more testing and `ping6 localhost` work on machine connected to corp but not my home local machine. So it seems like it depends on your environment.(e.g. name server configuration)  Also if you use 'localhost' there will be ambiguity about the address family as you may ask for QUIC_LOCALHOST_FOR_AF(QUIC_ADDRESS_FAMILY_INET6) but end up with IPv4 connection or vice versa. 

One more note that because this is string, it will still go through resolver logic to resolve it into binary address format.

Reply:
Let me know @nibanks how you want to proceed. We can fork this to separate issue as somewhat unrelated to the socket changes. 

I could pass all tests on my machine with 'loopback'. We may get lucky and that _may_ work for BVT as well. We just need to be aware of the fact that that is not generally available. (and the tests may not do what is intended) 

I could also roll-back this change and disable impacted tests for now.

Reply:
I don't care too much either way. Let's just get in something that works for now.

Reply:
ok. Using "::1" string passed BVT. I can try if "localhost" works for BVT. if not, I'll roll back to current state.

Reply:
macOS BVT passed with localhost so I'll leave as such.  I also check FreeBSD and ip6-localhost is not there either. So it really seems like Linux-ism more so than posix.

Reply:
> Its `ip6-localhost` not valid on macOS? Interesting, but not super shocking.